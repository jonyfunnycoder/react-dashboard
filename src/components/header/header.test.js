import React from 'react';
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import Avtar from "../../images/avtar.png";
import Header from "./header";

test('renders header correctly ', () => {
    const tree = renderer
        .create(<Header
            userName="Sunil Dandwate"
            userRole="Soft. Dev."
            avtarUrl={Avtar}
        />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test('renders header without avtar correctly ', () => {
    const tree = renderer
        .create(<Header
            userName="Sunil Dandwate"
            userRole="Soft. Dev."
        />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test('renders header without avtar correctly ', () => {
    const tree = renderer
        .create(<Header
            userName="Sunil Dandwate"
            userRole="Soft. Dev."
        />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test('render the contect correct ', () => {
    const { getByText, getByAltText } = render(<Header
        userName="Sunil Dandwate"
        userRole="Soft. Dev."
        avtarUrl={Avtar}
    />);

    getByText("Sunil Dandwate");
    getByText("Soft. Dev.");
    getByAltText("User");
});

test('render the header without avtar ', () => {
    const { getByText } = render(<Header
        userName="Sunil Dandwate"
        userRole="Soft. Dev."
    />);

    getByText("Sunil Dandwate");
    getByText("Soft. Dev.");
});
