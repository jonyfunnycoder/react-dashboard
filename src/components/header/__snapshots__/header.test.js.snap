// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header correctly  1`] = `
Array [
  <div
    className="header gutter-2x"
  >
    <nav
      className="navbar navbar-expand-md navbar-light bg-light"
    >
      <ul
        className="ml-auto navbar-nav"
      >
        <li
          className="dropdown nav-item"
          onKeyDown={[Function]}
        >
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle nav-link"
            href="#"
            onClick={[Function]}
          >
            <div
              className="user-dropdown"
            >
              <div
                className="user-info"
              >
                <div
                  className="user-name"
                >
                  Sunil Dandwate
                </div>
                <div
                  className="user-role"
                >
                  Soft. Dev.
                </div>
              </div>
              <img
                alt="User"
                className="avtar"
                src="avtar.png"
              />
            </div>
          </a>
          <div
            aria-hidden={true}
            className="dropdown-menu dropdown-menu-right"
            role="menu"
            tabIndex="-1"
          >
            <button
              className="dropdown-item"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
              type="button"
            >
              Menu #1
            </button>
            <div
              className="dropdown-divider"
              onClick={[Function]}
              tabIndex="-1"
            />
            <button
              className="dropdown-item"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
              type="button"
            >
              Menu #2
            </button>
            <div
              className="dropdown-divider"
              onClick={[Function]}
              tabIndex="-1"
            />
            <button
              className="dropdown-item"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
              type="button"
            >
              Menu #3
            </button>
          </div>
        </li>
      </ul>
    </nav>
  </div>,
  <div
    className="header-height"
  />,
]
`;

exports[`renders header without avtar correctly  1`] = `
Array [
  <div
    className="header gutter-2x"
  >
    <nav
      className="navbar navbar-expand-md navbar-light bg-light"
    >
      <ul
        className="ml-auto navbar-nav"
      >
        <li
          className="dropdown nav-item"
          onKeyDown={[Function]}
        >
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle nav-link"
            href="#"
            onClick={[Function]}
          >
            <div
              className="user-dropdown"
            >
              <div
                className="user-info"
              >
                <div
                  className="user-name"
                >
                  Sunil Dandwate
                </div>
                <div
                  className="user-role"
                >
                  Soft. Dev.
                </div>
              </div>
            </div>
          </a>
          <div
            aria-hidden={true}
            className="dropdown-menu dropdown-menu-right"
            role="menu"
            tabIndex="-1"
          >
            <button
              className="dropdown-item"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
              type="button"
            >
              Menu #1
            </button>
            <div
              className="dropdown-divider"
              onClick={[Function]}
              tabIndex="-1"
            />
            <button
              className="dropdown-item"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
              type="button"
            >
              Menu #2
            </button>
            <div
              className="dropdown-divider"
              onClick={[Function]}
              tabIndex="-1"
            />
            <button
              className="dropdown-item"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
              type="button"
            >
              Menu #3
            </button>
          </div>
        </li>
      </ul>
    </nav>
  </div>,
  <div
    className="header-height"
  />,
]
`;

exports[`renders header without avtar correctly  2`] = `
Array [
  <div
    className="header gutter-2x"
  >
    <nav
      className="navbar navbar-expand-md navbar-light bg-light"
    >
      <ul
        className="ml-auto navbar-nav"
      >
        <li
          className="dropdown nav-item"
          onKeyDown={[Function]}
        >
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle nav-link"
            href="#"
            onClick={[Function]}
          >
            <div
              className="user-dropdown"
            >
              <div
                className="user-info"
              >
                <div
                  className="user-name"
                >
                  Sunil Dandwate
                </div>
                <div
                  className="user-role"
                >
                  Soft. Dev.
                </div>
              </div>
            </div>
          </a>
          <div
            aria-hidden={true}
            className="dropdown-menu dropdown-menu-right"
            role="menu"
            tabIndex="-1"
          >
            <button
              className="dropdown-item"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
              type="button"
            >
              Menu #1
            </button>
            <div
              className="dropdown-divider"
              onClick={[Function]}
              tabIndex="-1"
            />
            <button
              className="dropdown-item"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
              type="button"
            >
              Menu #2
            </button>
            <div
              className="dropdown-divider"
              onClick={[Function]}
              tabIndex="-1"
            />
            <button
              className="dropdown-item"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
              type="button"
            >
              Menu #3
            </button>
          </div>
        </li>
      </ul>
    </nav>
  </div>,
  <div
    className="header-height"
  />,
]
`;
